src/App.tsx

```diff
@@ -1,24 +1,170 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import React, {
+  useState, useEffect, useMemo, useCallback,
+} from 'react';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { Todo } from './types/Todo';
+import {
+  getTodos, addTodo, removeTodo, updateTodo,
+} from './api/todos';
+import { FilterType } from './types/FilterType';
+import { getFilteredTodos } from './utils/getFilteredTodos';
+import { ErrorMessage } from './components/ErrorMessage';
+import { ErrorMessages } from './types/ErrorMessages';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterType, setFilterType] = useState(FilterType.All);
+  const [errorType, setErrorType] = useState<ErrorMessages>(ErrorMessages.None);
+  const [isErrorHidden, setIsErrorHidden] = useState(true);
+  const [updatingStage, setUpdatingStage] = useState<number[]>([]);
+  const [editedTodoId, setEditedTodoId] = useState(0);
+
+  const loadTodos = async () => {
+    try {
+      const loadedTodos = await getTodos();
+
+      setTodos(loadedTodos);
+    } catch (error) {
+      setErrorType(ErrorMessages.Load);
+      setIsErrorHidden(false);
+    }
+  };
+
+  const addToUpdateStage = (id: number) => {
+    setUpdatingStage(prev => [...prev, id]);
+  };
+
+  const removeFromUpdateStage = (id: number) => {
+    setTimeout(() => {
+      setUpdatingStage(prev => prev.filter(prevInd => prevInd !== id));
+    }, 500);
+  };
+
+  const filteredTodos = useMemo(() => {
+    return getFilteredTodos(todos, filterType);
+  }, [filterType, todos]);
+
+  useEffect(() => {
+    loadTodos();
+  }, []);
+
+  useEffect(() => {
+    setTimeout(setIsErrorHidden, 3000, true);
+  }, [isErrorHidden]);
+
+  const addTodoOnServer = useCallback(async (todo: Todo) => {
+    try {
```

Comment:
here is a great place to implement logic from previous  task 
![image](https://user-images.githubusercontent.com/58560038/220362351-f870a887-9980-437b-986a-959edbea30a4.png)